src/components/Modal/types.ts

```diff
@@ -4,7 +4,13 @@ import { fontSizeType } from "../../common/types"
 
 export interface ModalWrapperProps extends BaseComponentProps {
   /** Contents to be rendered in the header section */
-  header: { title: string; fontSize?: fontSizeType, rightSection?: ReactNode, noFill?: boolean }
+  header: {
```

Comment:
When the `leftSection` is not a string, then the other props become irrelevant, so I'm thinking we can let `leftSection` be either a `ReactNode` or an object containing the required properties (`title`, `fontSize` etc)

Reply:
this seems better API changes pushed in latest commit, @Kushagra8888